assets/js/app.js

```diff
@@ -0,0 +1,99 @@
+// Wait for the DOM to fully load before executing any JavaScript
+document.addEventListener("DOMContentLoaded", () => {
+    // Get references to the form and the task board where tasks will be displayed
+    const taskForm = document.getElementById("taskForm");
+    const taskBoard = document.getElementById("taskBoard");
+  
+    // Function to load tasks from localStorage and display them on the task board
+    const loadTasks = () => {
+        // Retrieve tasks from localStorage; if none exist, initialize an empty array
+        const tasks = JSON.parse(localStorage.getItem("tasks")) || [];
+        // Loop through each task and display it if it hasn't expired
+        tasks.forEach((task) => {
+            if (!isTaskExpired(task.date, task.time)) {
+                addTaskToBoard(task);
+            }
+        });
+    };
+  
+    // Function to save tasks currently displayed on the task board into localStorage
+    const saveTasks = () => {
+        const tasks = []; // Initialize an empty array to store task data
+        // Select all task cards on the board
+        document.querySelectorAll('.task-card').forEach(taskCard => {
+            // Extract task details from the task card's content
+            const description = taskCard.querySelector('.task-content').innerHTML;
+            const date = taskCard.querySelector('.due-date div:nth-child(1)').innerText.replace('Date: ', '');
+            const time = taskCard.querySelector('.due-date div:nth-child(2)').innerText.replace('Time: ', '');
+            // Add the task details to the tasks array
+            tasks.push({ description, date, time });
+        });
+        // Store the tasks array as a JSON string in localStorage
+        localStorage.setItem('tasks', JSON.stringify(tasks));
+    };
+  
+    // Function to check if a task's due date and time have already passed
+    const isTaskExpired = (date, time) => {
+        const taskDateTime = new Date(`${date}T${time}`); // Create a Date object from the task's date and time
+        const now = new Date(); // Get the current date and time
+        return taskDateTime < now; // Return true if the task's time has passed
+    };
+  
+    // Function to add a task to the task board
+    const addTaskToBoard = ({ description, date, time }) => {
+        if (isTaskExpired(date, time)) return;
```

Comment:
creates unclear UX. you add a task but nothing happens. no error message. also it makes it hard to later debug the removal of expired items